### ISSUE TITLE
Weird channel force-closure

### ISSUE THREAD

**Issue by bonomat** [get10101/10101#1426]
We had a channel force closure which looks like an HTLC timed out, but from what I can tell, there is no HTLC on the commitment transaction.

channel: `5ead0e9c4b98cfe8dd13fefcf6ac2ced32f96ce932e679d0dc0794f45c3b3dfa`
funding tx: [`fa3d3b5cf49407dcd079e632e96cf932ed2cacf6fcfe13dde8cf984b9c0ead5e`](https://mempool.space/tx/fa3d3b5cf49407dcd079e632e96cf932ed2cacf6fcfe13dde8cf984b9c0ead5e#vout=0)
closing tx: [`3085d6759087222ed9537cb485bd61802406c8e6f827ccfc02edd8cbbd987da8`](https://mempool.space/tx/3085d6759087222ed9537cb485bd61802406c8e6f827ccfc02edd8cbbd987da8)
claiming tx: [`3085d6759087222ed9537cb485bd61802406c8e6f827ccfc02edd8cbbd987da8`](https://mempool.space/tx/fc93212929d85a98034d583d1ca58eefb2d96b6392acd6b34a05fa75e08e3bff)
user pk: `03c1ffa91ff307a85fe6228bf3e3e6137ee531281ac6d33fa4659503623f37419b`
environment: production

Coordinator app

```
{"timestamp":"2023-09-25T10:07:57.60105217Z","level":"INFO","fields":{"message":"Force-closing channel due to outbound HTLC timeout, HTLC expiry is 809265","[log.target](http://log.target/)":"lightning::chain::channelmonitor","log.module_path":"lightning::chain::channelmonitor","log.file":"/home/runner/.cargo/git/checkouts/rust-lightning-bee3233422fa4387/3b69cec/lightning/src/chain/[channelmonitor.rs](http://channelmonitor.rs/)","log.line":3434},"target":"lightning::chain::channelmonitor"}
```

App log a bit later when the user came back online
```
2023-09-29 13:41:51.478  [LogLevel.INFO](http://loglevel.info/)  lightning::chain::channelmonitor: Force-closing channel due to inbound  HTLC timeout, HTLC expiry is 809265
```

**Comment by bonomat** [comment]
Chain of events: 

funding tx: https://mempool.space/tx/fa3d3b5cf49407dcd079e632e96cf932ed2cacf6fcfe13dde8cf984b9c0ead5e

closing tx: https://mempool.space/tx/3085d6759087222ed9537cb485bd61802406c8e6f827ccfc02edd8cbbd987da8

output spending tx: 
https://mempool.space/tx/fc93212929d85a98034d583d1ca58eefb2d96b6392acd6b34a05fa75e08e3bff

The last one is not an HTLC right?

**Comment by bonomat** [comment]
Payment hash of the JIT funding `eab09e95dc29327674891dfa91d1b85bc8ea62111b8b8fcf4ea5a06e7c5e050c`

**Comment by bonomat** [comment]
On the app side we successfully removed this htlc again

```
023-09-25 11:41:45.065  LogLevel.INFO  Received payment amount_msat: 100000000,payment_hash: 0xea0xb00x9e0x950xdc0x290x320x760x740x890x1d0xfa0x910xd10xb80x5b0xc80xea0x620x110x1b0x8b0x8f0xcf0x4e0xa50xa00x6e0x7c0x5e0x050x0c  
2023-09-25 11:41:45.071  LogLevel.TRACE  lightning::ln::channel: Upgrading HTLC eab09e95dc29327674891dfa91d1b85bc8ea62111b8b8fcf4ea5a06e7c5e050c to LocalRemoved with a Fulfill in channel 5ead0e9c4b98cfe8dd13fefcf6ac2ced32f96ce932e679d0dc0794f45c3b3dfa!   
2023-09-25 11:41:45.072  LogLevel.TRACE  lightning::ln::channel: Updating HTLC state for a newly-sent commitment_signed...   
2023-09-25 11:41:45.078  LogLevel.TRACE  lightning::ln::channel: Building commitment transaction number 281474976710653 (really 2 xor 2994835194686) for channel 5ead0e9c4b98cfe8dd13fefcf6ac2ced32f96ce932e679d0dc0794f45c3b3dfa for remote, generated by us with fee 6096...   
2023-09-25 11:41:45.080  LogLevel.TRACE  lightning::ln::channel:    ...not including inbound HTLC 0 (hash eab09e95dc29327674891dfa91d1b85bc8ea62111b8b8fcf4ea5a06e7c5e050c) with value 100000000 due to state (LocalRemoved)   
2023-09-25 11:41:45.081  LogLevel.TRACE  lightning::ln::channel:    ...including to_remote output with value 95587   
2023-09-25 11:41:45.082  LogLevel.TRACE  lightning::ln::channel:    ...including to_local output with value 100000   
2023-09-25 11:41:45.083  LogLevel.TRACE  lightning::ln::channelmanager: Tracking monitor update completion action for channel 5ead0e9c4b98cfe8dd13fefcf6ac2ced32f96ce932e679d0dc0794f45c3b3dfa: PaymentClaimed { payment_hash: PaymentHash([234, 176, 158, 149, 220, 41, 50, 118, 116, 137, 29, 250, 145, 209, 184, 91, 200, 234, 98, 17, 27, 139, 143, 207, 78, 165, 160, 110, 124, 94, 5, 12]) }   
2023-09-25 11:41:45.085  LogLevel.TRACE  lightning::chain::chainmonitor: Updating ChannelMonitor for channel 5ead0e9c4b98cfe8dd13fefcf6ac2ced32f96ce932e679d0dc0794f45c3b3dfa   
2023-09-25 11:41:45.087  LogLevel.INFO  lightning::chain::channelmonitor: Applying update to monitor 5ead0e9c4b98cfe8dd13fefcf6ac2ced32f96ce932e679d0dc0794f45c3b3dfa, bringing update_id from 2 to 3 with 2 changes.   
2023-09-25 11:41:45.088  LogLevel.TRACE  lightning::chain::channelmonitor: Updating ChannelMonitor with payment preimage   
2023-09-25 11:41:45.090  LogLevel.TRACE  lightning::chain::channelmonitor: Updating ChannelMonitor with latest counterparty commitment transaction info   
2023-09-25 11:41:45.093  LogLevel.TRACE  lightning::chain::channelmonitor: Tracking new counterparty commitment transaction with txid 3085d6759087222ed9537cb485bd61802406c8e6f827ccfc02edd8cbbd987da8 at commitment number 281474976710653 with 0 HTLC outputs   
2023-09-25 11:41:45.094  LogLevel.DEBUG  lightning::chain::chainmonitor: Persistence of ChannelMonitorUpdate for channel 5ead0e9c4b98cfe8dd13fefcf6ac2ced32f96ce932e679d0dc0794f45c3b3dfa completed   
2023-09-25 11:41:45.095  LogLevel.TRACE  lightning::ln::channel: Regenerated latest commitment update in channel 5ead0e9c4b98cfe8dd13fefcf6ac2ced32f96ce932e679d0dc0794f45c3b3dfa with 0 update_adds, 1 update_fulfills, 0 update_fails, and 0 update_fail_malformeds   
2023-09-25 11:41:45.096  LogLevel.TRACE  lightning::ln::channel: Building commitment transaction number 281474976710653 (really 2 xor 2994835194686) for channel 5ead0e9c4b98cfe8dd13fefcf6ac2ced32f96ce932e679d0dc0794f45c3b3dfa for remote, generated by us with fee 6096...   
2023-09-25 11:41:45.097  LogLevel.TRACE  lightning::ln::channel:    ...not including inbound HTLC 0 (hash eab09e95dc29327674891dfa91d1b85bc8ea62111b8b8fcf4ea5a06e7c5e050c) with value 100000000 due to state (LocalRemoved)   
2023-09-25 11:41:45.099  LogLevel.TRACE  lightning::ln::channel:    ...including to_remote output with value 95587   
2023-09-25 11:41:45.100  LogLevel.TRACE  lightning::ln::channel:    ...including to_local output with value 100000   
2023-09-25 11:41:45.101  LogLevel.TRACE  lightning::ln::channel: Signed remote commitment tx 02000000015ead0e9c4b98cfe8dd13fefcf6ac2ced32f96ce932e679d0dc0794f45c3b3dfa00000000004ab90280026375010000000000220020e3460e67d515d246fee6be5a7887c3fc24d786905f34d7e33d037c232295cc02a08601000000000016001417de5aede2abeb450f4ea11146a132503b16e21a3c831620 (txid 3085d6759087222ed9537cb485bd61802406c8e6f827ccfc02edd8cbbd987da8) with redeemscript 4752210292e9e07760bf87566889442d9b1d156a3b04472ecf41fda0b8fd6b99eb23801e21039cd31fa3814b7e9860fa55e1099c8edd55826dfab9e8afd280c54fce447f42eb52ae with value 200000 -> 1409752d2d30c03707f41d52e70938c456089018f43286adc29d478fd62b0aea683b877500d19a7e50d19973be22134390bd36e4a9622692484a46ea91ee3574 in channel 5ead0e9c4b98cfe8dd13fefcf6ac2ced32f96ce932e679d0dc0794f45c3b3dfa   
2023-09-25 11:41:45.102  LogLevel.DEBUG  lightning::ln::channel: Restored monitor updating in channel 5ead0e9c4b98cfe8dd13fefcf6ac2ced32f96ce932e679d0dc0794f45c3b3dfa resulting in a commitment update and no RAA, with RAA first   
2023-09-25 11:41:45.103  LogLevel.TRACE  lightning::ln::channelmanager: Handling channel resumption for channel 5ead0e9c4b98cfe8dd13fefcf6ac2ced32f96ce932e679d0dc0794f45c3b3dfa with no RAA, a commitment update, 0 pending forwards, not broadcasting funding, without channel ready, without announcement   
2023-09-25 11:41:45.104  LogLevel.DEBUG  Successfully handled event event: "PaymentClaimable { receiver_node_id: Some(PublicKey(9b41373f62039565a43fd3c61a2831e57e13e6e3f38b22e65fa807f31fa9ffc1d7e326d9eb5d87273a9572066e52c4a6b735cf32509d593976d777595445e3de)), payment_hash: PaymentHash([234, 176, 158, 149, 220, 41, 50, 118, 116, 137, 29, 250, 145, 209, 184, 91, 200, 234, 98, 17, 27, 139, 143, 207, 78, 165, 160, 110, 124, 94, 5, 12]), amount_msat: 100000000, purpose: InvoicePayment { payment_preimage: Some(PaymentPreimage([78, 28, 161, 38, 31, 201, 71, 156, 21, 24, 214, 188, 40, 198, 215, 36, 169, 154, 196, 17, 7, 26, 84, 74, 3, 212, 120, 132, 130, 192, 163, 163])), payment_secret: PaymentSecret([123, 204, 186, 177, 73, 109, 40, 58, 224, 149, 43, 163, 253, 65, 139, 66, 135, 125, 121, 41, 222, 188, 183, 122, 71, 198, 185, 214, 68, 193, 104, 98]) }, via_channel_id: Some([94, 173, 14, 156, 75, 152, 207, 232, 221, 19, 254, 252, 246, 172, 44, 237, 50, 249, 108, 233, 50, 230, 121, 208, 220, 7, 148, 244, 92, 59, 61, 250]), via_user_channel_id: Some(56299488804169558210290719508778983382) }"  
2023-09-25 11:41:45.105  LogLevel.INFO  Received event event: PaymentClaimed { receiver_node_id: Some(PublicKey(9b41373f62039565a43fd3c61a2831e57e13e6e3f38b22e65fa807f31fa9ffc1d7e326d9eb5d87273a9572066e52c4a6b735cf32509d593976d777595445e3de)), payment_hash: PaymentHash([234, 176, 158, 149, 220, 41, 50, 118, 116, 137, 29, 250, 145, 209, 184, 91, 200, 234, 98, 17, 27, 139, 143, 207, 78, 165, 160, 110, 124, 94, 5, 12]), amount_msat: 100000000, purpose: InvoicePayment { payment_preimage: Some(PaymentPreimage([78, 28, 161, 38, 31, 201, 71, 156, 21, 24, 214, 188, 40, 198, 215, 36, 169, 154, 196, 17, 7, 26, 84, 74, 3, 212, 120, 132, 130, 192, 163, 163])), payment_secret: PaymentSecret([123, 204, 186, 177, 73, 109, 40, 58, 224, 149, 43, 163, 253, 65, 139, 66, 135, 125, 121, 41, 222, 188, 183, 122, 71, 198, 185, 214, 68, 193, 104, 98]) } }  
2023-09-25 11:41:45.106  LogLevel.INFO  Claimed payment amount_msat: 100000000,payment_hash: eab09e95dc29327674891dfa91d1b85bc8ea62111b8b8fcf4ea5a06e7c5e050c  
2023-09-25 11:41:45.107  LogLevel.DEBUG  Getting payment payment_hash: eab09e95dc29327674891dfa91d1b85bc8ea62111b8b8fcf4ea5a06e7c5e050c  
2023-09-25 11:41:45.108  LogLevel.INFO  Updating payment payment_hash: PaymentHash([234, 176, 158, 149, 220, 41, 50, 118, 116, 137, 29, 250, 145, 209, 184, 91, 200, 234, 98, 17, 27, 139, 143, 207, 78, 165, 160, 110, 124, 94, 5, 12])  
2023-09-25 11:41:45.109  LogLevel.DEBUG  Successfully handled event event: "PaymentClaimed { receiver_node_id: Some(PublicKey(9b41373f62039565a43fd3c61a2831e57e13e6e3f38b22e65fa807f31fa9ffc1d7e326d9eb5d87273a9572066e52c4a6b735cf32509d593976d777595445e3de)), payment_hash: PaymentHash([234, 176, 158, 149, 220, 41, 50, 118, 116, 137, 29, 250, 145, 209, 184, 91, 200, 234, 98, 17, 27, 139, 143, 207, 78, 165, 160, 110, 124, 94, 5, 12]), amount_msat: 100000000, purpose: InvoicePayment { payment_preimage: Some(PaymentPreimage([78, 28, 161, 38, 31, 201, 71, 156, 21, 24, 214, 188, 40, 198, 215, 36, 169, 154, 196, 17, 7, 26, 84, 74, 3, 212, 120, 132, 130, 192, 163, 163])), payment_secret: PaymentSecret([123, 204, 186, 177, 73, 109, 40, 58, 224, 149, 43, 163, 253, 65, 139, 66, 135, 125, 121, 41, 222, 188, 183, 122, 71, 198, 185, 214, 68, 193, 104, 98]) } }"  
2023-09-25 11:41:45.110  LogLevel.DEBUG  lightning::ln::peer_handler: Handling UpdateHTLCs event in peer_handler for node 022ae8dbec1caa4dac93f07f2ebf5ad7a5dd08d375b79f11095e81b065c2155156 with 0 adds, 1 fulfills, 0 fails for channel 5ead0e9c4b98cfe8dd13fefcf6ac2ced32f96ce932e679d0dc0794f45c3b3dfa   
2023-09-25 11:41:45.111  LogLevel.TRACE  lightning_background_processor: Persisting ChannelManager...   
2023-09-25 11:41:45.112  LogLevel.TRACE  lightning_background_processor: Done persisting ChannelManager.   
```

**Comment by bonomat** [comment]
Here is the coordinator preparing the closing transaction: [Grafana link](https://itchysats.grafana.net/goto/e7V6OiGIR?orgId=1)

**Comment by bonomat** [comment]
Ok, here my conclusiong:

It looks like we lost connection to the user.

It's a bit of a weird scenario: 
We received the preimage from the user, including their signed commitment tx, successfully claimed the HTLC backwards. 
Then lost connection to the user and did not update our internal state. 

We then thought we did not remove the HTLC (even though we did remove it? ) and force closed using THEIR commitment transaction. 

Yes, WE published the commitment transaction which had THEIR money locked in a timelock but not ours. 

That's a bit weird to me.
